qutip/core/data/kron.pyx

```diff
@@ -159,6 +177,22 @@ cpdef Dia kron_dia(Dia left, Dia right):
     return out
 
 
+cpdef Dia kron_dia_dense_dia(Dia left, Dense right):
+    # The dispatcher would use kron_dense, but the output is at least as sparse
+    # as the sparse input. Since the dispatcher does not have precise control
+    # on which function to use when the signature is missing. We add
+    # then like this.
```

Comment:
just a suggestion here; how about "Since the dispatcher does not have precise control on which function to use when the signature is missing, we specify the output to be the sparse format of the input."  or something like that (the original is a bit opaque)